hugolib/pagemeta/page_frontmatter.go

```diff
@@ -59,6 +60,16 @@ type FrontMatterDescriptor struct {
 	// May be set from the author date in Git.
 	GitAuthorDate time.Time
 
+	// This the titlefunc used by page site
+	// todo: would have preferred to point to Site to read other configuration
+	// but cannot use hugolib from nested package
+	TitleFunc func(s string) string
```

Comment:
Would have been nicer to be able to pass `Site` in to let handlers  lookup config or in this case `titleFunc` but package `hugolib/pagemeta` cannot use `hugolib` due to go package nesting limits.